### ISSUE TITLE
Type Literal Reference with recursion

### ISSUE THREAD

**Issue by uittorio** [uittorio/ts-auto-mock#88]
this should compile correctly.

```ts 
interface A {
    a: this;

    b: {
        c: string;
    }
}
```

It gives an error on the compiler.

**Reason:**
The current descriptor for typeLiteral is acting as an InterfaceDeclaration.
Because of this it will replace the current this node causing any other this reference to fail.

**Solution:**
We should treat typesLiteral as a different descriptor that will not store the this value.

 #83

**Comment by uittorio** [comment]
#83

**Comment by uittorio** [comment]
This issue has been resolved in version 1.3.1
